partitioner/src/main/java/io/confluent/connect/storage/partitioner/FieldPartitioner.java

```diff
@@ -17,55 +17,76 @@
 
 public class FieldPartitioner<T> extends DefaultPartitioner<T> {
   private static final Logger log = LoggerFactory.getLogger(FieldPartitioner.class);
-  private String fieldName;
+  private List<String> fieldNames;
 
   @Override
   public void configure(Map<String, Object> config) {
-    fieldName = (String) config.get(PartitionerConfig.PARTITION_FIELD_NAME_CONFIG);
-    partitionFields = newSchemaGenerator(config).newPartitionFields(fieldName);
+    fieldNames = (List<String>) config.get(PartitionerConfig.PARTITION_FIELD_NAME_CONFIG);
     delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
+
+    StringBuilder format = new StringBuilder();
+    for (String field : fieldNames) {
+      format.append(field + delim);
+    }
+
+    if (format.length() > 0) {
+      format.delete(format.length() - delim.length(), format.length());
+    }
+
+    partitionFields = newSchemaGenerator(config).newPartitionFields(format.toString());
   }
 
   @Override
   public String encodePartition(SinkRecord sinkRecord) {
     Object value = sinkRecord.value();
     Schema valueSchema = sinkRecord.valueSchema();
+    StringBuilder partition = new StringBuilder();
+
     if (value instanceof Struct) {
       Struct struct = (Struct) value;
-      Object partitionKey = struct.get(fieldName);
-      Type type = valueSchema.field(fieldName).schema().type();
-      switch (type) {
-        case INT8:
-        case INT16:
-        case INT32:
-        case INT64:
-          Number record = (Number) partitionKey;
-          return fieldName + "=" + record.toString();
-        case STRING:
-          return fieldName + "=" + (String) partitionKey;
-        case BOOLEAN:
-          boolean booleanRecord = (boolean) partitionKey;
-          return fieldName + "=" + Boolean.toString(booleanRecord);
-        default:
-          log.error("Type {} is not supported as a partition key.", type.getName());
-          throw new PartitionException("Error encoding partition.");
+      for (String fieldName : fieldNames) {
+        Object partitionKey = struct.get(fieldName);
+        Schema.Type type = valueSchema.field(fieldName).schema().type();
+        switch (type) {
+          case INT8:
+          case INT16:
+          case INT32:
+          case INT64:
+            Number record = (Number) partitionKey;
+            partition.append(fieldName + "=" + record.toString() + delim);
+            break;
+          case STRING:
+            partition.append(fieldName + "=" + (String) partitionKey + delim);
+            break;
+          case BOOLEAN:
+            boolean booleanRecord = (boolean) partitionKey;
+            partition.append(fieldName + "=" + Boolean.toString(booleanRecord) + delim);
+            break;
+          default:
+            log.error("Type {} is not supported as a partition key.", type.getName());
+            throw new PartitionException("Error encoding partition.");
+        }
+      }
+
+      if (partition.length() > 0) {
```

Comment:
Again, this check won't be necessary if you use `org.apache.kafka.common.utils.Utils.join`